api/src/org/apache/cloudstack/api/command/user/vm/DeployVMCmd.java

```diff
@@ -333,6 +333,7 @@ public boolean getStartVm() {
                     } catch (NumberFormatException e) {
                         throw new InvalidParameterValueException("Unable to translate and find entity with networkId: " + ips.get("networkid"));
                     }
+                    // so no network found but a valid number as network id; are we assuming exceptions thrown or lucky guessing?
```

Comment:
@DaanHoogland so you are the one that makes the enlightened comments all around ACS code base?! ahahaha

If the network ID is invalid (it was not found), it means that something else further down the deployment process will break, right? If this is the case, I think we could stop the processing here and throw a runtime exception.

Reply:
I don't know yet, maybe an uncaught CloudRuntimeException is thrown somewhere else, but you are right, we can throw it here.

Reply:
Let's remove comments, unless they're around newly added code or explain existing code?

Reply:
sure, this comment should not add value any more once we decide how to handle this case.
At this point we have not found a network but are having a networkId. I think this should be unreachable.
I will refactor untill the code makes sense to me.